source/ops/touch.py

```diff
@@ -110,27 +102,15 @@ class VIEW2D_OT_TouchInput(Operator):
         options={"HIDDEN"},
     )
 
-    def execute(self, context: Context):
-        if context.area.type == "IMAGE_EDITOR":
-            return self.exec_image_editor(context)
-        if self.mode == "DOLLY":
-            bpy.ops.view2d.zoom("INVOKE_DEFAULT")  # type: ignore
-        elif self.mode == "PAN":
-            bpy.ops.view2d.pan("INVOKE_DEFAULT")  # type: ignore
-        return FINISHED
-
-    def exec_image_editor(self, _: Context):
-        if self.mode == "PAN":
-            bpy.ops.image.view_pan("INVOKE_DEFAULT")  # type: ignore
-        elif self.mode == "DOLLY":
-            bpy.ops.image.view_zoom("INVOKE_DEFAULT")  # type: ignore
-        return FINISHED
+    @classmethod
+    def poll(cls, context):
+        return context.area.type in {"NODE_EDITOR", "VIEW_2D", "IMAGE_EDITOR"} and context.region.type == "WINDOW"
 
-    def invoke(self, context: Context, event: Event):
-        settings = get_settings()
-        if not settings.is_enabled:
+    def invoke(self, context, event):
+        prefs = preferences()
+        if not prefs.is_enabled:
             return PASSTHROUGH
-        if settings.input_mode == "both" and (event.type == PEN or not isTouch(event)):
+        if prefs.input_mode == "both" and (event.type == PEN or not is_touch(event)):
```

Comment:
I think there is typo, `input_mode` doesn't have `both` as item.
You should fix it.